File: acmicpc/acmicpc_9243.cs
// 9243 파일 완전 삭제

using System.Diagnostics.Metrics;

namespace ConsoleApp1;

public class Program
{
    public static void Main(string[] args)
    {
        int n = Convert.ToInt32(Console.ReadLine());
        string before = Console.ReadLine();
        string after = Console.ReadLine();

        if (isSucess(n, before, after))
        {
            Console.WriteLine("Deletion succeeded");
        }
        else
        {
            Console.WriteLine("Deletion failed");
        }
    }

    private static bool isSucess(int n, string before, string after)
    {
        // 짝수만큼 뒤집으면 결과는 똑같음
        if (n % 2 == 0)
        {
            if (before == after)
            {
                return true;
            }
            else
            {
                return false;
            }
        }
        // 홀수만큼 뒤집으면 모든 값들이 반대
        else
        {
            for (int i = 0; i < before.Length; i++)
            {
                if (before[i] == after[i])
                {
                    return false;
                }
            }

            return true;
        }
    }
}

File: Python/acmicpc-py/acmicpc_9243.py
# 9243 파일 완전 삭제

import sys
input = sys.stdin.readline

n = int(input())
s1 = input()[:-1]
s2 = input()[:-1]

if n % 2 == 0:
  if s1 == s2:
    print("Deletion succeeded")
  else:
    print("Deletion failed")
else:
  s3 = ""
  for s in s1:
    s3 = s3 + ('1' if s == '0' else '0')
    
  if s2 == s3:
    print("Deletion succeeded")
  else:
    print("Deletion failed")

